### DISCUSSION TITLE
Limit visibility of certain printers

### DISCUSSION THREAD

**Discussion by forrie** [OpenPrinting/cups#654] [Ideas]
I'd like the ability to deploy printers in a manner that could limit visibility -- it could be IP network groups, users, etc.   In a larger organization, this could be potentially useful (accounting, resources, etc).    

Would this be reasonable for CUPS?

**Comment by michaelrsweet** [comment]
@forrie You can define user/group access lists which will limit access, but since *discovery* is necessarily anonymous it doesn't always work as you would expect (i.e. you might see and choose a printer only to discover you can't print to it).

Similarly, DNS-SD may not expose the same network constraints as the CUPS server - a client might find a record for a given printer but when it tries to access the printer it can be denied.
